Consolidate YAML files and update README

scholzj opened the pull request:
There is no need to have the YAML files split across several places and have duplicate content in them. This PR creates new top level dir `resources` where we concentrate all our YAML files (except the S2I YAML files which are not yet implemented). The build system has been updated to reflect this change.

Additionally, this PR:
* Allows to specify cluster name
* Fix labels in Kuberenetes controller YAML file
* updates the README file to reflect Cluster Controller and the new location of the files.

ppatierno commented on README.md:
```diff
@@ -57,205 +56,136 @@ oc cluster up
-* config : configuration file templates for running Zookeeper
-* scripts : scripts for starting up Kafka and Zookeeper servers
-* resources : provides all YAML configuration files for setting up volumes, services and deployments
+Strimzi is using process called Cluster Controller to deploy and manage Kafka and Kafka Connect clusters. Cluster Controller is a process which is running inside your Kubernetes or OpenShift cluster. To deploy a Kafka cluster, a ConfigMap with the cluster configuration has to be created. The ConfigMap needs ot be labaled with followign labels:
```
do we want to add something like "with related Zookeeper ensemble" or something like that ?

ppatierno replied:
There are also three typos in a row, congratulations !! :-) "...needs ot be labaled with followign ..." instead of " ... needs to be labeled with following ..."

ppatierno commented on README.md:
```diff
@@ -57,205 +56,136 @@ oc cluster up
+* regular services can be used as bootstrap servers for Kafka clients;
+* headless services are needed to have DNS resolve the pods IP addresses directly.
 
+Strimzi provides two flavors of Kafka broker deplyoment: **ephemeral** and **persisitent**. 
```
typo "persisitent" ...

ppatierno commented on README.md:
```diff
@@ -57,205 +56,136 @@ oc cluster up
-
-        oc new-app strimzi
+Kafka broker is provided in the form of OpenShift template. The cluster can be deployed from the template either 
+using command line or using the OpenShift console. To create the ephemeral cluster, run following commend in your 
```
typo "commend" ...

scholzj commented:
@ppatierno Yeah, it looks like I forgot to run it through spellchecker. I fixed the typos you noticed and some more. I also added the Zookeeper reference.

scholzj commented:
@ppatierno Yeah, that might make sense. The Docker image it self is now named `kafka-statefulsets` as the directory. Would you rename that as well? Or keep the image name and change only the directory?

ppatierno commented:
@scholzj right ! I would rename the Docker image name as well. Finally we don't need to have "statefulsets" in any of the named stuff imho.

scholzj commented:
@ppatierno Well, the Docker image is dependent on stateful sets (it takes the broker ID from its hostname). But we do not have any other image, so we do not need to have it in the name. We probably should have done it before the 0.1.0 release, but I guess it makes sense to do it and the sooner we do it the better.

ppatierno commented:
@scholz we are using statefulsets but it's actually the only tecnology that gives us the possibility to deploy a stateful application like Kafka so having it in the name sounds redundant to me even after your great refactoring. In any case the 0.1.0 release will be replaced by newer releases completely different in terms of implementation. For me this release is just a "temporary" one. :)

scholzj commented:
@ppatierno Agreed ... it is already implemented.
